services/vidispine/chapter.go

```diff
@@ -0,0 +1,217 @@
+package vidispine
+
+import (
+	"fmt"
+	"math"
+	"regexp"
+	"strings"
+
+	"github.com/orsinium-labs/enum"
+	"github.com/samber/lo"
+)
+
+type ChapterType enum.Member[string]
+
+var (
+	ChapterTypeSong      = ChapterType{"song"}
+	ChapterTypeSpeech    = ChapterType{"speech"}
+	ChapterTypeTestimony = ChapterType{"testimony"}
+	ChapterTypeOther     = ChapterType{"other"}
+	ChapterTypeCredits   = ChapterType{"credits"}
+
+	ChapterTypeMap = map[string]ChapterType{
+		"sang":            ChapterTypeSong,
+		"musikkvideo":     ChapterTypeSong,
+		"musikal":         ChapterTypeSong,
+		"tale":            ChapterTypeSpeech,
+		"appelle":         ChapterTypeSpeech,
+		"vitnesbyrd":      ChapterTypeTestimony,
+		"end-credit":      ChapterTypeCredits,
+		"panel":           ChapterTypeOther,
+		"intervju":        ChapterTypeOther,
+		"temafilm":        ChapterTypeOther,
+		"animation":       ChapterTypeOther,
+		"programleder":    ChapterTypeOther,
+		"dokumentar":      ChapterTypeOther,
+		"ordforklaring":   ChapterTypeOther,
+		"frsending":       ChapterTypeOther,
+		"ettersending":    ChapterTypeOther,
+		"bildekavaalkade": ChapterTypeOther,
+		"skuespill":       ChapterTypeOther,
+		"aksjonstatus":    ChapterTypeOther,
+		"hilse":           ChapterTypeOther,
+		"konkuranse":      ChapterTypeOther,
+		"informasjon":     ChapterTypeOther,
+		"bnn":             ChapterTypeOther,
+		"promo":           ChapterTypeOther,
+		"mte":             ChapterTypeOther,
+		"fest":            ChapterTypeOther,
+		"underholdning":   ChapterTypeOther,
+		"kortfilm":        ChapterTypeOther,
+		"anslag":          ChapterTypeOther,
+		"teaser":          ChapterTypeOther,
+		"reality":         ChapterTypeOther,
+		"studio":          ChapterTypeOther,
+		"talk-show":       ChapterTypeOther,
+		"presentasjon":    ChapterTypeOther,
+		"seminar":         ChapterTypeOther,
+		"reportasje":      ChapterTypeOther,
+	}
+)
+
+var SongExtract = regexp.MustCompile("(FMB|HV) ?-? ?([0-9]+)")
+var SongCollectionMap = map[string]string{
+	"FMB": "AB",
+	"HV":  "WOTL",
+}
+
+type Chapter struct {
+	ChapterType    string
+	Timestamp      float64
+	Label          string
+	Title          string
+	Description    string
+	SongCollection string
+	SongNumber     string
+	Highlight      bool
+	Persons        []string
+}
+
+func (c *Client) GetChapterData(exportData *ExportData) ([]Chapter, error) {
+	metaCache := map[string]*MetadataResult{}
+
+	allChapters := map[string]*MetadataResult{}
+
+	for _, clip := range exportData.Clips {
+		if _, ok := metaCache[clip.VXID]; !ok {
+			meta, err := c.GetMetadata(clip.VXID)
+			if err != nil {
+				return nil, err
+			}
+			metaCache[clip.VXID] = meta
+		}
```

Comment:
I think the vidispine client is initialized per activity/request, so this cache could probably be stored on the client struct and moved out?

Reply:
Yes I wanted it to be at most cached for one request.